README.md
# Simple Black

A black colorscheme for (neo)vim, real black background, #000000.

Color schemes define colors for syntax groups which are defined by Vim and syntax highlighting plugins. This colorscheme should work even if you don't have the syntax highlighting plugins listed below.

Supported languages/syntaxes:
- Ruby using [vim-ruby/vim-ruby](https://github.com/vim-ruby/vim-ruby)
- JavaScript using [yuezk/vim-js](https://github.com/yuezk/vim-js)
- TypeScript using [HerringtonDarkholme/yats.vim](https://github.com/HerringtonDarkholme/yats.vim)
- Elixir using [elixir-editors/vim-elixir](https://github.com/elixir-editors/vim-elixir)

More on the way!

![simpleblack](github/simpleblack.png)

## Install

Using [Plug](https://github.com/junegunn/vim-plug) and getting updates when running `:PlugUpdate`.

```
Plug 'lucasprag/simpleblack'
```

You can also set a specific version if you want to not receive updates.

```
Plug 'lucasprag/simpleblack', { 'tag': 'v0.4.0' } " latest version
```

## Screenshots

#### Ruby

Notice that you can distinguish between module, class and method endings by color.

![ruby](github/ruby.png)

#### JavaScript

![javascript](github/javascript.png)

#### TypeScript

![typescript](github/typescript.png)

#### Elixir

![elixir](github/elixir.png)

## Usage

Change your `~/.vimrc` or `~/.config/nvim/init.vim`.

```
colorscheme simpleblack
```

## Inspiration
- [irblack](https://github.com/wesgibbs/vim-irblack)
- [SpaceCamp](https://github.com/jaredgorski/SpaceCamp)

Thanks


CHANGELOG.md
# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [0.4.0] - 2020-05-24
### Added
- Support for Elixir.
- Add color to `nil` in Ruby.

## [0.3.0] - 2020-05-24
### Added
- Support for TypeScript.

## [0.2.0] - 2020-05-23
### Added
- Support for JavaScript.

## [0.1.0] - 2020-05-23
### Added
- Support for Ruby.
- Changelog and semantic versioning using git tags.


github/ruby.rb
require 'active_support'

module VimColors
  class RubyExample
    CONSTANT = /^[0-9]+ regex awesomes$/

    attr_reader :colorscheme

    # TODO: Bacon ipsum dolor sit amet
    def initialize(attributes = {})
      @colorscheme = attributes[:colorscheme]
    end

    def self.examples
      # Bacon ipsum dolor sit amet
      ['string', :symbol, true, false, nil, 99.9, 1..2].each do |value|
        puts "it appears that #{value.inspect} is a #{value.class}"
      end

      {:key1 => :value1, key2: 'value2'}.each do |key, value|
        puts "the #{key.inspect} key has a value of #{value.inspect}"
      end

      %w[One Two Three].each { |number| puts number }
    end

    private

    def heredoc_example
      <<-SQL
        SELECT *
        FROM colorschemes
        WHERE background = 'dark'
      SQL
    end
  end
end




examples/elixir.ex
require "vimcolors"

defmodule ElixirExample do
  use Vim.Colors

  @moduledoc """
  Shows syntax highlithing for creating a screenshot
  """

  defmodule User do
    defstruct email: nil, role: "guest"
  end

  @doc false
  def examples(_args) do
    ["string", :atom, true, false, 99.9]
    |> Enum.each(fn(item) -> IO.puts("Look! We got a #{item}!") end)

    map = %{key:  <<1::3>>}
    if Map.has_key?(map, :key) do
      IO.inspect(%{map | key: "new value"})
    end

    Enum.min([~D[2020-03-31], ~D[2020-04-01]])
    Enum.min_by(["a", "aa", "aaa", "b", "bbb"], &String.length/1)

    [key: "value", key: "different value"]
  end

  def message(user = %User{email: ~r/example/}) do
    case user.role do
      "admin"  -> "Hey admin."
      "member" -> "Hello again member, please text the admin for help if you need."
      "guest"  -> "Welcome guest, feel free to sign up to be a member."
      _        -> "I don't know who you are, but be my guest."
    end
  end
end


examples/javascript.js
import { VimColors } from "vimcolors"

class JavaScriptExample extends VimColors {
  constructor(attributes = {}) {
    this.colorscheme = attributes.colorscheme;
  }

  static examples() {
    // Bacon ipsum dolor sit amet
    ['string', true, false, null, 99.9, 42, function(){}].forEach(function(value) {
      console.log(`it appears that ${value} is a ${typeof value}`);
    })

    Object.entries({string: 'value', number: 2}).forEach((key, value) => {
      console.warn(`the ${key} key has a value of ${typeof value}`);
    })
  }

  present(options) {
    if (this.colorscheme.match(/dark/)) {
      return "My colorscheme called " + this.colorscheme + " is dark.";
    }

    super(options);
  }
}

var example = new JavaScriptExample({ colorscheme: "simpleblack" });
document.getElementById("demo").innerHTML = example.present();

let changeMe = "42"
const cantChangeMe = 42
window.alert(changeMe)
console.log(changeMe = cantChangeMe, changeMe)




examples/typescript.ts
import { VimColors } from "vimcolors"

export class TypeScriptExample extends VimColors {
  constructor(public colorscheme: string) {
    super(arguments);
    this.colorscheme = colorscheme;
  }

  static example(): Array<number> {
    // Bacon ipsum dolor sit amet
    let list: Array<number> = [1, 2, 3];

    ['string', true, false, null, 99.9, 42, function(){}].forEach(function(value) {
      console.log(`it appears that ${value || "null" } is a ${typeof value}`);
    })

    Object.entries({string: 'value', number: 2}).forEach((key: string, value: any) => {
      console.warn(`the ${key} key has a value of ${typeof value}`);
    })

    return list;
  }

  public present(): string {
    if (this.colorscheme.match(/dark/)) {
      return "My colorscheme called " + this.colorscheme + " is dark.";
    }
  }
}

interface Example {
  colorscheme: string;
}

function greeter(example: Example) {
  return "Hello, " + example.colorscheme;
}

var example = new TypeScriptExample("simpleblack");
document.getElementById("demo").innerHTML = example.present();

window.alert(greeter(example));

let changeMe: string = "42"
const cantChangeMe: number = 42
console.log(changeMe = "new value", changeMe, cantChangeMe)



